src/ast/mod.rs

```diff
@@ -122,21 +121,59 @@ impl From<&str> for Ident {
     fn from(value: &str) -> Self {
         Ident {
             value: value.to_string(),
-            quote_style: None,
+            quote_style: QuoteStyle::None,
         }
     }
 }
 
 impl fmt::Display for Ident {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         match self.quote_style {
-            Some(q) if q == '"' || q == '\'' || q == '`' => {
-                let escaped = value::escape_quoted_string(&self.value, q);
-                write!(f, "{}{}{}", q, escaped, q)
+            QuoteStyle::None => f.write_str(&self.value),
+            QuoteStyle::SingleQuote | QuoteStyle::DoubleQuote | QuoteStyle::BackTick => {
+                let escaped = value::escape_quoted_string(&self.value, self.quote_style.as_char());
+                write!(f, "{}{}{}", self.quote_style, escaped, self.quote_style)
+            }
+            QuoteStyle::SquareBracket => write!(f, "[{}]", self.value),
+        }
+    }
+}
+
+/// Valid quoting characters
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Copy)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub enum QuoteStyle {
+    /// No quote
+    None,
+    /// Quote with `'`
+    SingleQuote,
+    /// Quote with `"`
+    DoubleQuote,
+    /// Quote with ``` (BackTick)
+    BackTick,
+    /// Quote with `[]` (Square bracket)
+    SquareBracket,
+}
+impl fmt::Display for QuoteStyle {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            QuoteStyle::SingleQuote => write!(f, "'"),
+            QuoteStyle::DoubleQuote => write!(f, "\""),
+            QuoteStyle::BackTick => write!(f, "`"),
+            QuoteStyle::SquareBracket | QuoteStyle::None => Ok(()),
+        }
+    }
+}
+
+impl QuoteStyle {
+    fn as_char(&self) -> char {
+        match self {
+            QuoteStyle::SingleQuote => '\'',
+            QuoteStyle::DoubleQuote => '"',
+            QuoteStyle::BackTick => '`',
+            QuoteStyle::SquareBracket | QuoteStyle::None => {
+                panic!("Not supported for quote style {}", self)
```

Comment:
@omer-shtivi please make this an error. Panicking here isn't nice at all.

Reply:
Done